tests/manage/test_pvc_delete_verify_size_is_returned_to_backendpool.py

```diff
@@ -0,0 +1,178 @@
+        f"pvc {pv_name} exists on backend"
+    )
+    logger.info(
+        f"Expected: pvc {pv_name} doesn't exist on backend after deleting pvc"
+    )
+
+
+def create_pvc_and_verify_pvc_exists(
+        sc_name, cbp_name,
+        desired_status=constants.STATUS_BOUND, wait=True
+):
+    """
+    Create pvc, verify pvc is bound in state and
+    pvc exists on ceph side
+    """
+
+    pvc_data = defaults.CSI_PVC_DICT.copy()
+    pvc_data['metadata']['name'] = helpers.create_unique_resource_name(
+        'test', 'pvc'
+    )
+    pvc_data['spec']['storageClassName'] = sc_name
+    pvc_data['spec']['resources']['requests']['storage'] = "10Gi"
+    global PVC_OBJ
+    PVC_OBJ = helpers.create_resource(**pvc_data, wait=False)
+    assert PVC_OBJ, (
+        f"Failed to create resource {pvc_data['metadata']['name']}"
+    )
+    if wait:
+        assert PVC_OBJ.ocp.wait_for_resource(
+            condition=desired_status, resource_name=PVC_OBJ.name
+        ), f"{pvc_data.kind} {PVC_OBJ.name} failed to reach"
+        f"status {desired_status}"
+
+    # get pvc info
+    pvc_info = PVC_OBJ.get()
+    pv_name = pvc_info['spec']['volumeName']
+
+    # Validate pvc is created on ceph
+    logger.info(f"Verifying pvc {pv_name} exists on backend")
+    _rc = False
+    ct_pod = pod.get_ceph_tools_pod()
+    pvc_list = ct_pod.exec_ceph_cmd(
+        ceph_cmd=f"rbd ls -p {cbp_name}", format='json'
+    )
+    for pvc in pvc_list:
+        if pvc == pv_name:
+            _rc = True
+            break
+    assert _rc, f"pvc {pv_name} doesn't exist on backend"
+    logger.info(f"pvc {pv_name} exists on backend")
+    return pv_name, PVC_OBJ.name
+
+
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__
+)
+class TestPVCDeleteAndVerifySizeIsReturnedToBackendPool(ManageTest):
+    """
+    Testing after pvc deletion the size is returned to backendpool
+    """
+
+    @tier1
+    def test_pvc_delete_and_verify_size_is_returned_to_backend_pool(self):
+        """
+        Test case to verify after delete pvc size returned to backend pools
+        """
+        used_before_creating_pvc = check_ceph_used_space()
+        logger.info(f"used before creating pvc {used_before_creating_pvc}")
+        pv_name, pvc_name = create_pvc_and_verify_pvc_exists(
+            self.sc_obj.name, self.cbp_obj.name
+        )
+        pod_data = defaults.CSI_RBD_POD_DICT.copy()
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
+        pod_obj = helpers.create_pod(**pod_data)
+        run_io(pod_obj)
+        used_after_creating_pvc = check_ceph_used_space()
+        logger.info(f"used after creating pvc {used_after_creating_pvc}")
+        assert used_before_creating_pvc < used_after_creating_pvc
+        pod_obj.delete()
+        PVC_OBJ.delete()
+        # Todo: Takes some time to delete pv so for now adding sleep
+        time.sleep(20)
```

Comment:
what we can sample to check if this was ended sooner than 20 seconds?

Reply:
done